### PR TITLE
Add tests of ServerResultUtils#validateResult to make sure an empty body is sent for 1xx, 204 and 304 responses

### PR DESCRIPTION
**Pull request by monkey-mas**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you [squashed your commits]? (Optional, but makes merge messages nicer.)
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Purpose
Currently there's no tests to check if the returned body of 1xx, 204 and 304 responses is empty, which is required by RFC7230. To make sure that no entity is sent to a client, we can add unit tests of `ServerResultUtils#validateResult`[1] that takes responsibility for cancelling a body if necessary.

## Background Context
PR(#6261)[2] was merged to meet the requirement of RFC7230. `BasicHttpClient` is used for the integration tests to check if an empty body is returned for 1xx, 204 and 304 responses. These integration tests, however, don't guarantee that those responses return an empty body as our `BasicHttpClient` just returns an empty body instead of sending an actual body.[3]

## References
PR(#6261)[2]
PR Comment[4]

[1] https://github.com/playframework/playframework/blob/2.5.4/framework/src/play-server/src/main/scala/play/core/server/common/ServerResultUtils.scala#L67-L68
[2] https://github.com/playframework/playframework/pull/6261
[3] https://github.com/playframework/playframework/blob/2.5.4/framework/src/play-integration-test/src/test/scala/play/it/http/BasicHttpClient.scala#L214-L218
[4] https://github.com/playframework/playframework/pull/6261#issuecomment-235596925

### CODE REVIEW

File: framework/src/play-server/src/test/scala/play/core/server/common/ServerResultUtilsSpec.scala [hunk 1]
```diff
@@ -68,4 +76,87 @@ class ServerResultUtilsSpec extends Specification {
       }
     }
   }
+
+  "ServerResultUtils#validateResult" should {
+    val app = GuiceApplicationBuilder().build()
+    implicit val materializer = app.materializer
```
Comment (gmethvin) [inline framework/src/play-server/src/test/scala/play/core/server/common/ServerResultUtilsSpec.scala]: You don't actually need an application here. You could just use
```scala
implicit val system = ActorSystem()
implicit val materializer = ActorMaterializer()
```

File: framework/src/play-server/src/test/scala/play/core/server/common/ServerResultUtilsSpec.scala [hunk 2]
```diff
@@ -68,4 +76,87 @@ class ServerResultUtilsSpec extends Specification {
...
+
+    val header = new RequestHeader {
```
Comment (gmethvin) [inline framework/src/play-server/src/test/scala/play/core/server/common/ServerResultUtilsSpec.scala]: FakeRequest?

**Comment by gmethvin** [comment]
lgtm

File: framework/src/play-server/src/test/scala/play/core/server/common/ServerResultUtilsSpec.scala (hunk 2 above)
Comment (monkey-mas) [inline framework/src/play-server/src/test/scala/play/core/server/common/ServerResultUtilsSpec.scala]: You mean `play.core.test.FakeRequest`? To use it, I think we need to modify [PlayProject](https://github.com/playframework/playframework/blob/2.5.4/framework/project/Build.scala#L273) in `Build.scala#PlayServerProject` to `PlayProject % "compile->compile;test->test"`. I agree to use `FakeRequest` but I'm not sure if it's ok to add `test->test` dependency just for this test. WDYT? :eyes:

Comment (gmethvin) [inline framework/src/play-server/src/test/scala/play/core/server/common/ServerResultUtilsSpec.scala]: Okay, since it's just for this test don't worry about it.

**Comment by monkey-mas** [comment]
@gmethvin : Fixed and rebased the first commit :)
